jenkins-integration/beaker/install/shared/45_install_hiera_config.rb

```diff
@@ -0,0 +1,39 @@
+require 'puppet/gatling/config'
+require 'tmpdir'
+require 'yaml'
+
+test_name 'Install hiera config'
+
+def target_hiera_config_path(host)
+  return on(host, puppet('config print hiera_config')).stdout.chomp
+end
+
+def get_source_hiera_config(host, hiera_config)
+  source_hiera_config_path = hiera_config[:source_file]
+  if source_hiera_config_path.nil? || source_hiera_config_path.empty?
+    source_hiera_config_path = target_hiera_config_path(host)
```

Comment:
Yeah, you're right that it would overwrite the file with exactly the same content if the variables were both empty.  I thought about putting some more conditional logic in to skip doing anything for that case but had thought the code read a little more cleanly without doing that.  I (or somebody else I guess) could still add that back in if desirable later.